modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/quantity_selector/QuantitySelector.js

```diff
@@ -12,246 +12,122 @@
  * details.
  */
 
+import {ClaySelect} from '@clayui/form';
 import ClayIcon, {ClayIconSpriteContext} from '@clayui/icon';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
-import React, {createRef, useEffect, useState} from 'react';
+import React, {useEffect, useState} from 'react';
 
-import throttle from '../../utilities/throttle';
-
-const THROTTLE_TIMEOUT = 1000;
+import Datalist from '../datalist/Datalist';
+import {generateOptions, getInputStyle} from './utils/index';
 
 function QuantitySelector(props) {
-	const [currentQuantity, setCurrentQuantity] = useState(
-		!!props.minQuantity && props.minQuantity > props.quantity
-			? props.minQuantity
-			: props.quantity
-	);
-	const [nextAvailable, setNextAvailable] = useState(
-		currentQuantity + props.multipleQuantity <= props.maxQuantity
-	);
-	const [prevAvailable, setPrevAvailable] = useState(
-		currentQuantity - props.multipleQuantity >= props.minQuantity
-	);
-	const [isThrottling, setIsThrottling] = useState(false);
+	const [selectedQuantity, setSelectedQuantity] = useState(props.quantity);
+
+	const quantitiesList = generateOptions(props.settings);
+
+	const updateQuantity = (q) => {
+		if (q !== null) {
+			setSelectedQuantity(q);
+			try {
+				props.updateQuantity(q);
+			}
+			catch (e) {
+				// eslint-disable-next-line no-console
+				console.log('stand alone component');
+			}
+		}
+	};
 
-	const inputRef = createRef();
+	const inputStyle = getInputStyle(props.inputSize);
 
 	useEffect(() => {
-		setCurrentQuantity(props.quantity);
-	}, [props.quantity, setCurrentQuantity]);
-
-	useEffect(() => {
-		if (props.throttleOnUpdate) {
-			setIsThrottling(true);
-
-			props.onUpdate(currentQuantity).then(() => setIsThrottling(false));
-		}
-		else {
-			props.onUpdate(currentQuantity);
-		}
+		// eslint-disable-next-line no-unused-expressions
```

Comment:
Please don't override the linter. Rewrite using statements, which is what it wants you to do:

```js
if (props.quantity) {
    updateQuantity(props.quantity);
}
else {
    updateQuantity(selectedQuantity);
}
```

Or:

```js
updateQuantity(props.quantity ? props.quantity : selectedQuantity);
```

Or:

```js
updateQuantity(props.quantity || selectedQuantity);
```

Of those, the last is probably my favorite because it is succinct.